hack/tools/confdocs/confdocs.go

```diff
@@ -542,3 +543,122 @@ func doInitLogging(level string) {
 
 	logger = l.Sugar()
 }
+
+var (
```

Comment:
Let's put the code extracted from `structtag` into its own file and include the copyright notice from https://github.com/fatih/structtag/blob/master/LICENSE.